src/install/lockfile/Package.zig

```diff
@@ -1088,7 +1088,7 @@ pub const Package = extern struct {
             .npm => {
                 const npm = dependency_version.value.npm;
                 if (workspace_version != null) {
-                    if (npm.version.satisfies(workspace_version.?, buf, buf)) {
+                    if (npm.version.satisfies(workspace_version.?, buf, buf) and pm.options.link_workspace_packages) {
```

Comment:
```suggestion
                    if (pm.options.link_workspace_packages and npm.version.satisfies(workspace_version.?, buf, buf)) {
```

we can skip `satisfies` if linkWorkspacePackages == false